src/utils/vcs.rs

```diff
@@ -249,15 +249,41 @@ fn find_matching_rev(
         if let Some(url) = remote.url();
         then {
             if !discovery || is_matching_url(url, &reference_url) {
-                debug!("  found match: {} == {}", url, &reference_url);
+                debug!("  found match: {} == {}, {:?}", url, &reference_url, r);
                 let head = repo.revparse_single(r)?;
+                if let Some(tag) = head.as_tag(){
+                    debug!("  tag.target(): {:?}", tag.target());
+                    if let Ok(tag_commit) = tag.target() {
+                        return Ok(Some(log_match!(tag_commit.id().to_string())));
+                    }
+                }
+                debug!("  not a tag");
+
                 return Ok(Some(log_match!(head.id().to_string())));
             } else {
                 debug!("  not a match: {} != {}", url, &reference_url);
             }
         }
     }
+    if let Ok(submodule_match) = find_matching_submodule(r, reference_url, repo) {
+        return Ok(submodule_match);
+    }
+    info!("  -> no matching revision found");
+    Ok(None)
+}
 
+fn find_matching_submodule(
```

Comment:
Was this factored out for a specific reason or just to restructure?

If you prefer to keep it that way, please move the macro out of the functions to prevent duplication.

Reply:
I took it out because the linter was saying the function was too cognitively complex (31). 
Macro function moved out :)